ui/component/navigationButton/view.jsx

```diff
@@ -0,0 +1,75 @@
+// @flow
+import React, { useState, useCallback } from 'react';
+import * as ICONS from 'constants/icons';
+import Button from 'component/button';
+
+type Props = {
+  isBackward: boolean,
+  history: {
+    entries: { key: string, title: string }[],
+    go: number => void,
+    goBack: () => void,
+    goForward: () => void,
+    index: number,
+    length: number,
+    location: { pathname: string },
+    push: string => void,
+  },
+};
+
+// determines which slice of entries should make up the back or forward button drop-downs (isBackward vs !isBackward respectively)
+const sliceEntries = (currentIndex, entries, historyLength, isBackward) => {
+  const l = isBackward ? 0 : currentIndex + 1;
+  const r = isBackward ? currentIndex : historyLength;
+  return entries.slice(l, r);
+};
+
+const NavigationButton = (props: Props) => {
+  const { isBackward, history } = props;
+  const { entries, go } = history;
+  const currentIndex = history.index;
+  const historyLength = history.length;
+  const [showHistory, setShowHistory] = useState(false);
+
+  // creates an <li> intended for the button's <ul>
+  const makeItem = useCallback(
+    (entry, index) => {
+      // difference between the current index and the index of the entry
+      const goArg = isBackward ? index - currentIndex : index + 1;
+      return (
+        <li
+          key={entry.key}
+          onMouseDown={() => {
+            setShowHistory(false);
+            go(goArg);
+          }}
+        >
+          {entry.title}
+        </li>
+      );
+    },
+    [currentIndex, isBackward, go]
+  );
+  const slicedEntries = sliceEntries(currentIndex, entries, historyLength, isBackward);
+  return (
+    <div>
+      <Button
```

Comment:
Are you able to determine if we can go back here or not? If so, it would be great to disable the back button and give it some disabled style so it's easier to tell that you can't

Reply:
PR has been updated to disable button if no historical entries.

Reply:
Awesome!